### PR TITLE
Boilerplate Updates

### PR DESCRIPTION
**Pull request by khamer**
Here's the updates to Boilerplate itself that I've done thus far. Let's keep most visual update/design changes separate, so we can talk about that separately.

Feedback encouraged - once I get enough positive feedback, I'll merge this in and replace Boilerplate as it exists right now so we can use this updates immediately. (Any sass lib stuff we'll do separately when it's ready.)

### CODE REVIEW

**Comment by ghost** [comment]
please update the author meta tag (line 10) to be web_author:

<meta name="web_author" content="Imarc, A Digital Agency | Boston + Silicon
Valley | more info at imarc.com" />

source: http://handbook.imarc.net/frontend

On Tue, Apr 12, 2016 at 1:54 PM, Kevin Hamer notifications@github.com
wrote:

> Here's the updates to Boilerplate itself that I've done thus far. Let's
> keep most visual update/design changes separate, so we can talk about that
> separately.
> 
> Feedback encouraged - once I get enough positive feedback, I'll merge this
> in and replace Boilerplate as it exists right now so we can use this
> updates immediately. (Any sass lib stuff we'll do separately when it's
> 
> ## ready.)
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/imarc/boilerplate/pull/41
> Commit Summary
> - updating css to 1.2
> - changing whitespace to spaces from tabs
> - ran boilerplate through CSSComb
> - getting rid of in favor of rem
> - changing to two levels of colors
> - removing bg-gradient
> - improved fontawesome mixin
> - removing no bullets
> - putting no bullets back
> - cleaning up section dividers; removing icon section
> - updating toc
> - Merging in 2016 changes
> - Switching from tabs to spaces
> - switching to specs preferred double colon for psuedo elements
> - switching to using a CDN to provide fontawesome
> - shouldn't have both link-states() and link-hover; dropping
>   link-states() mixin in favor of defining link-hover and link-visited as
>   variables
> - no need to use named, lt-ie9 compatible media queries mixins anymore
> - fixing a typo
> 
> File Changes
> - _A_ .csscomb.json
>   https://github.com/imarc/boilerplate/pull/41/files#diff-0 (32)
> - _D_ css/font-awesome-lite.css
>   https://github.com/imarc/boilerplate/pull/41/files#diff-1 (196)
> - _M_ css/styles.css
>   https://github.com/imarc/boilerplate/pull/41/files#diff-2 (795)
> - _M_ css/styles.css.map
>   https://github.com/imarc/boilerplate/pull/41/files#diff-3 (4)
> - _M_ css/styles.scss
>   https://github.com/imarc/boilerplate/pull/41/files#diff-4 (1554)
> - _M_ index.html
>   https://github.com/imarc/boilerplate/pull/41/files#diff-5 (1633)
> - _M_ js/bp.js
>   https://github.com/imarc/boilerplate/pull/41/files#diff-6 (58)
> - _M_ structure.html
>   https://github.com/imarc/boilerplate/pull/41/files#diff-7 (490)
> 
> Patch Links:
> - https://github.com/imarc/boilerplate/pull/41.patch
> - https://github.com/imarc/boilerplate/pull/41.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/imarc/boilerplate/pull/41

## 

[image: iMarc Logo]

_Shawna O'Neal_ • Jr. Web Engineer • Imarc, A Digital Agency
shawna@imarc.com victoria@imarc.com • 978.462.8848

**Comment by mbfitz** [comment]
Re: "I can't decide where to put definitions for classes like .grouping, .container, .columns, .tiles, etc." 
I think the 3rd option is the best - maybe create a separate section for "Components", add it to the TOC, and add the components classes in this section. This way, we can keep what we need, and strip out what we don't need per project. Having the class within the corresponding doc block in the components.scss file would be nice, too, in case we need to go back and add one we stripped out.

File: .csscomb.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "color-case": "upper",
+  "color-shorthand": true,
+  "element-case": "lower",
+  "eof-newline": true,
+  "leading-zero": true,
+  "quotes": "double",
+  "remove-empty-rulesets": true,
+  "space-after-combinator": " ",
+  "space-after-opening-brace": "\n",
+  "space-after-selector-delimiter": " ",
```
Comment (TommyChanthaboune) [inline .csscomb.json:13]: Is this what is causing the closing brace to be on the same line as the last property ?

Comment (TommyChanthaboune) [inline .csscomb.json:13]: Wait nvm I was looking at the compiled css

**Comment by TommyChanthaboune** [comment]
+1

**Comment by marcelmoreau** [comment]
1. can we remove the /fonts/directory now 
2. can we show an icon example that uses our new fontawesome mixin?

**Comment by khamer** [comment]
I removed the fonts directory. As for the icon example, I nuked all of the existing icon SCSS since we weren't using it on any site. Any suggestion for what SCSS would make sense to include in boilerplate? it is part of the messaging mixin definition, but it doesn't show passing the character as an argument.

At worst, when we transition to components, then we'll probably be able to add standalone examples that aren't part of boilerplate, but if there's something you have in mind to add to boilerplate now let me know.

**Comment by ghost** [comment]
Might as well update the iMarc instances to Imarc while you're working on this (index.html)

**Comment by TommyChanthaboune** [comment]
This branch has already been merged in master. Lets open up an issue if there are questions or feature requests to keep track of it all and make it discoverable for other contributors.

**Pull request merged**